### PR TITLE
Add WalletList and WalletDetail Views

### PR DESCRIPTION
**Pull request by chedug**
Now it's possible to view all wallets using "wallets/" endpoint, and view details of each wallet using "wallets/<str:name>" endpoints.

### CODE REVIEW

File: requirements.txt
Comment (VladKli) [inline requirements.txt:1]: I do not see django, DRF packages. Could you double check if all necessary dependencies are in the file?

Comment (chedug) [reply requirements.txt:1]: Oh, sorry, didn't notice that the comment is about requirements.txt. I will add it later, thank you for pointing that out.

File: testproject/wallets/models.py
Comment (VladKli) [inline testproject/wallets/models.py:1]: - split wallets and transactions into different applications.
Wallet:
1. function `create_random_name()` move to the class
2. id is unnecessary field 
3. for fields type and currency set default values
4. created_on,  modified_on null-False

Transactions:
1. for sender, receiver change related_name, and set attr to_field=name
2. set transfer_amount to 0.00 
3. commission, timestamp editable=False
4. status set to default

File: testproject/wallets/serializers.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+from .models import Transaction, Wallet
+
+
+class WalletSerializer(serializers.ModelSerializer):
+    """
+    Serializer for Wallet Model
+    """
+
+    user = serializers.ReadOnlyField(source="user.username")
```
Comment (VladKli) [inline testproject/wallets/serializers.py:18]: check what's PrimaryKeyRelatedField and compare which one is better to use

File: testproject/wallets/serializers.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        bonus = 0.00
+
+        if currency == "GBP":
+            bonus = Decimal(100.00)
+        elif currency in ("USD", "EUR"):
+            bonus = Decimal(3.00)
+
+        balance = validated_data.get("balance")
+        if balance is None:
+            balance = 0.00
```
Comment (VladKli) [inline testproject/wallets/serializers.py:67]: check this condition, please

File: testproject/transactions/serializers.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        sender = validated_data["sender"]
+        receiver = validated_data["receiver"]
+        sender_wallet = Wallet.objects.get(name=sender.name)
+        receiver_wallet = Wallet.objects.get(name=receiver.name)
+        if sender_wallet.currency != receiver_wallet.currency:
+            raise serializers.ValidationError(
+                "Wallets' currencies do not match."
+            )
+
+    def validate_transfer_amount(self, validated_data):
```
Comment (chedug) [inline testproject/transactions/serializers.py:72]: Error: When making POST request, validated_data consists of only Decimal("0") value (default value for transfer_amount).

Comment (chedug) [reply testproject/transactions/serializers.py]: Fixed error by renaming function to validate_sender_money (How?)

File: testproject/wallets/serializers.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+            "timestamp",
+        ]
+
+
+class UserSerializer(serializers.ModelSerializer):
+    """
+    Serializer for default auth.User model
+    """
+
+    wallets = serializers.RelatedField(many=True, read_only=True)
```
[the remark this answers is not in the document]
Comment (chedug) [reply testproject/wallets/serializers.py]: Done

File: testproject/wallets/serializers.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        self.validate_max_wallets(validated_data)
+        self.validate_bonus(validated_data)
+        return Wallet.objects.create(**validated_data)
+
+    def validate_max_wallets(self, validated_data):
+        """
+        User cannot have more than 5 wallets
+        """
+        user = validated_data["user"]
+        if Wallet.objects.filter(user=user).count() >= 5:
```
[the remark this answers is not in the document]
Comment (chedug) [reply testproject/wallets/serializers.py:52]: Done

**Review by VladKli — CHANGES_REQUESTED** [review]
1. urls
2. create constant variables for number of wallet, bonuses, etc. use UPPER_CASE
3. remove UserSerializer as we don't use it. correct urls and views accordingly
4. views. attrs should be at the beginning of a class then methods
5. WalletDetail(generics.RetrieveUpdateDestroyAPIView). we need only retrieve, delete
6. permissions
7. logic.py -> utils.py

**Review by VladKli — CHANGES_REQUESTED** [review]
1. add endpoints related to authorization to the registration app, login, logout. correct UserRegistrationView (it shouldn't return 500 error when user doesn't provide username or any other field during registration)
2. pre-commit
3. add types annotations
4. remove unused code

**Review by VladKli — APPROVED** [review]

**Pull request merged**